### PR TITLE
Add support for adding OpenAPI schema for GET requests with a body

### LINKED ISSUE #713: [FEATURE] Support body in GET and other methods with undefined behavior
**Describe the bug**
In new version of rest api specification get methods can have body but fastapi not add it to swagger spec.

**To Reproduce**
Create get method with some body parameters.

**Expected behavior**
Body parameters present in the specification.

**Environment:**
 - OS: Windows
 - FastAPI Version: 0.42.0

- Python version, get it with: 3.8

### PR DESCRIPTION
**Pull request by victorphoenix3**
Related to #713 .

### CODE REVIEW

**Comment by victorphoenix3** [comment]
@IamCathal could you take a look at this?

File: tests/test_get_request_body_schema.py [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+class Product(BaseModel):
+    name: str
+    description: str = None
+    price: float
+
+
+@app.get("/product")
+async def create_item(product: Product):
+    pass
```
Comment (rkbeatss) [inline tests/test_get_request_body_schema.py:16]: i think this is where the coverage is missing so try adding `# pragma: nocover` here

Comment (victorphoenix3) [reply tests/test_get_request_body_schema.py]: hey thanks @rkbeatss . It worked!

**Comment by victorphoenix3** [comment]
@tiangolo The PR is ready for review.

File: tests/test_get_request_body_schema.py (hunk 1 above)
Comment (tiangolo) [reply tests/test_get_request_body_schema.py]: Thanks for the help @rkbeatss ! :nerd_face:

**Comment by tiangolo** [comment]
Awesome @victorphoenix3 ! :rocket: :tada: 

I updated the test to check that it actually can receive a body in a `GET` request and to keep the same style of testing OpenAPI from other tests.

Thanks for your contribution! :cake: :tada:

**Pull request merged**